### PR TITLE
Assignment 1 Classification 1

### PR DESCRIPTION
**Pull request by NikuX**

### CODE REVIEW

**Comment by axrd** [comment]
# Module 01 - Logistic Regression & One-Hot Encoding  

**Overall**: Well done! You're accuracy score is pretty darn good 👍 
**Feedback**: Comment your code! Comment it to force a good habit of discipline. Future you will likely not remember what you meant with every line of code in past notebooks. Keep him in mind. Also, use a validation set, and leverage your coefficients!

## Exploring/Cleaning data
- Try the `df.info()` method. Might help you as an initial step.
- It's easier to see the shape with `df.shape` instead of re-using the `len()` function
- Here's a way to make your code a bit more functional in regards to finding missing values:
```
def missing_values(data):
   total = data.isnull().sum()
   percent = data.isnull().sum().sort_values(ascending = False)/len(data)*100
   tp = pd.concat([total, percent], axis=1, keys=['Total', '%'])
   return(np.transpose(tp))
```
- Remember though, missing values are not only Nulls or NaNs. Read the links at the top of the Lecture notebook for more info. Look for excessive zeros for example! 

## Train/Val/Test Split
- 👍 

## One-Hot Encoding & Scaling
- Nicely done! Kudos for evaluating cardinality of your features👍 

## Logistic Regression Model
 - Yep. That's how you do it 👍 

## Coefficients
- Try and make the plot a bit more legible. 

## Submit to Kaggle Competition
- Aww yeah! How's it feel to be on the leaderboard?

**Comment by axrd** [comment]
# Module 02 - Decision Trees  

**Overall**: You used the Classifier correctly. Congratulations on breaking 80% Accuracy! 📈 
**Feedback**: **_Comment your code!_** Good job implementing a `wrangle(X)` function ⚡️ 

## More Exploring/Cleaning data
- Dig deeper. Lookout for missing or incorrect values in other Features. 

## Implement a Decision Tree Classifier
- Yep 👍


# Plot feature importances
- Nicely done 👍 

## Submit New Predictions to Kaggle
- I see you climbing the leaderboard 🙌

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Nicolae_Dubenco
## Assignment: Random_Rorests


# Part 1. Imports

***
**What went well:**

-Good collective use of imports. Nice and clean:

```python

%matplotlib inline
import category_encoders as ce
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import RobustScaler
from sklearn.tree import DecisionTreeClassifier
pd.set_option('display.float_format', '{:.2f}'.format)

```
# Import and read CSVs
-IMPRESSIVE MERGE :) and collection of data from Google Drive : 

```python

# Merge train_feartures.csv & train_labels.csv
train = pd.merge(pd.read_csv('https://drive.google.com/uc?export=download&id=14ULvX0uOgftTB2s97uS8lIx1nHGQIB0P'),
                pd.read_csv('https://drive.google.com/uc?export=download&id=1r441wLr7gKGHGLyPpKauvCuUOU556S2f'))

# Read test_features.csv & sample_admission.csv
test = pd.read_csv('https://drive.google.com/uc?export=download&id=1wvsYl9hbRbZuIuoaLWCsW_kbcxCdocHz')
sample_submission = pd.read_csv('https://drive.google.com/uc?export=download&id=1kfJewnmhowpUo381oSn3XqsQ6Eto23XV')

# Split train into train & val
train, val = train_test_split(train, train_size=.80, test_size=.20, 
                             stratify=train['status_group'], random_state=42)

                          


```
# Exploring categorical features
-
-Great use of [.select_dtypes](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.select_dtypes.html) to return a subset of the dataFrame’s columns:

```python
def identify_double_columns_data(X):
  """Identify the columns that contain the same data in the data set,
  type of data categorical"""
  for column in X.select_dtypes(exclude='number').columns:
    if X[column].nunique() <= 30:
      print('--------' * 10)
      print('feature displayed: >>>', column, ' <<<')
      print('                      ', len(column) * '-')
      print('value     &    times present in column')
      print(X[column].value_counts())



```


# Random Forest Classifier


- Good import makes things more relevant to people looking at your code for the first time : 


```python
%%time
from sklearn.ensemble import RandomForestClassifier
```


- Shows understanding of [.OneHotEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) and [.RandomForestClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html) : 


```python
pipeline = make_pipeline(
    ce.OneHotEncoder(use_cat_names=True), 
    SimpleImputer(strategy='mean'), 
    RandomForestClassifier(n_estimators=200, random_state=42, n_jobs=-1)
)

# Fit on train, score on val
model = pipeline.fit(X_train, y_train)
print('Validation Accuracy', pipeline.score(X_val, y_val))
```


# Stretch Goals

## XGBoost Classifier


- Beyond the material we are on. I'm insanly impreseed. Great job. : 

```python
model = XGBClassifier()
model.fit(X_train_encoded, y_train)
```
# What to work on
- [x] Comment more in code
- [x] Come to PM for questions or help
